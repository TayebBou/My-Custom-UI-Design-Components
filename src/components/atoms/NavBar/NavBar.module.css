.navbar{
    position: relative;
    top: 0;
    width: calc(100% - 90px);
    z-index: 999;
    padding: 1em 60px 1em 30px;
    display: flex;
}