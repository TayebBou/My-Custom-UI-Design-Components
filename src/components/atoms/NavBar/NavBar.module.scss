.navbar{
    position: relative;
    top: 0;
    width: calc(100% - 90px);
    z-index: 499;
    padding: 16px 60px 16px 30px;
    display: flex;
}