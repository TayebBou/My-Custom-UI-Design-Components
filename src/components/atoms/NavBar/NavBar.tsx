import { FC } from 'react'
import styles from './NavBar.module.css'

type NavBarProps = {
  children: React.ReactNode
  fixed?: boolean
  className?: string
}

const NavBar: FC<NavBarProps> = (props) => {
  const { children, fixed, className } = props

  return (
    <div
      className={`${styles.navbar} ${className}`}
      style={fixed ? { position: 'fixed' } : {}}
    >
      {children}
    </div>
  )
}

NavBar.defaultProps = {
    fixed: false,
    className: ''
}

export default NavBar
