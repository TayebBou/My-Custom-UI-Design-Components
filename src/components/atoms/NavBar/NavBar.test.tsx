import { render } from "@testing-library/react";
import NavBar from "./NavBar";

const props = {
  fixed: true,
  className: "NavBar__test",
};

describe("NavBar component", () => {
  test("should render correctly with props", () => {
    // Arrange
    const { asFragment } = render(<NavBar {...props}>test</NavBar>);
    // Assert
    expect(asFragment()).toMatchSnapshot();
  });

  test("should render correctly without props", () => {
    // Arrange
    const { asFragment } = render(<NavBar>test</NavBar>);
    // Assert
    expect(asFragment()).toMatchSnapshot();
  });
});
