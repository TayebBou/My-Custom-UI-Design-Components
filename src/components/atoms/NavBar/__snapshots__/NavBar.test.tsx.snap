// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component should render correctly with props 1`] = `
<DocumentFragment>
  <div
    class="navbar NavBar__test"
    style="position: fixed;"
  >
    test
  </div>
</DocumentFragment>
`;

exports[`NavBar component should render correctly without props 1`] = `
<DocumentFragment>
  <div
    class="navbar "
  >
    test
  </div>
</DocumentFragment>
`;
